launchdarkly-server-sdk/src/data_source.rs

```diff
@@ -173,20 +173,19 @@ impl DataSource for StreamingDataSource {
                                 }
                             },
                             Some(Err(e)) => {
-                                error!("error on event stream: {:?}", e);
-
-                                match e {
-                                    es::Error::Eof => {
-                                        continue;
-                                    }
-                                    _ => {
-                                        debug!("unhandled error; break");
-                                        break;
-                                    }
-                                }
+                                error!("error on event stream: {:?}; assuming event stream will reconnect", e);
```

Comment:
nit: If we are expecting to recover it might be better to have this as a `warn!` instead, but I'm going to merge this as is for now.